src/mmda/queriers/pdfminer_outline_querier.py

```diff
@@ -0,0 +1,272 @@
+    """Raised when a destination pointer cannot be parsed to a location."""
+
+
+class PDFNoUsableDestinations(PDFMinerOutlineParserError):
+    """Raised when there are no usable destinations for any outline metadata."""
+
+
+class PDFDestinationLocationMissing(PDFMinerOutlineParserError):
+    """Raised when there is no destination data on a particular outline item."""
+
+
+class PDFNoOutlines(PDFMinerOutlineParserError):
+    """Raised when underlying PDF contains no outline metadata (common)."""
+
+
+class PDFDestinationNotFound(PDFMinerOutlineParserError):
+    """Raised when the underlying destination cannot be found in the PDF."""
+
+
+class PDFInvalidSyntax(PDFMinerOutlineParserError):
+    """Raised when the PDF cannot be properly parsed."""
+
+
+class PDFPageMissing(PDFMinerOutlineParserError):
+    """Raised when the destination points to a page that cannot be found."""
+
+
+class PDFEncryptionError(PDFMinerOutlineParserError):
+    """Raised for encrypted PDFs."""
+
+
+class PDFRecursionError(PDFMinerOutlineParserError):
+    """Underlying parse has cyclic reference of some sort causing recursion error."""
+
+
+class PDFAttributeError(PDFMinerOutlineParserError):
+    """Wraps an underlying AttributeError."""
+
+
+class PDFTypeError(PDFMinerOutlineParserError):
+    """Wraps an underlying TypeError."""
+
+
+class PDFEOF(PDFMinerOutlineParserError):
+    """Raised when encountering an unexpected EOF in parsing."""
+
+
+class PDFPSSyntaxError(PDFMinerOutlineParserError):
+    """Raised if the PDF syntax is invalid."""
+
+
+@dataclass
+class _PDFDestination:
+    """Destination on the target page."""
+
+    top: float
+    left: float
+
+
+_OutlineMetadataKeys = Union[str, int, float]
+
+
+@dataclass
+class OutlineMetadata:
+    """A pointer to a section location in a PDF."""
+
+    id: int  # pylint: disable=invalid-name
+    title: str
+    level: int
+    page: int
+    l: float  # pylint: disable=invalid-name
+    t: float  # pylint: disable=invalid-name
+
+    @classmethod
+    def from_metadata_dict(
+        cls, metadata_dict: Dict[str, _OutlineMetadataKeys]
+    ) -> "OutlineMetadata":
+        """Instantiate from a metadata dict on a Document
+
+        Args:
+            metadata_dict (Dict[str, _OutlineMetadataKeys]): Document metadata object
+
+        Returns:
+            OutlineMetadata: Rehydrated OutlineMetadata object
+        """
+        return OutlineMetadata(
+            id=metadata_dict["id"],
+            title=metadata_dict["title"],
+            level=metadata_dict["level"],
+            page=metadata_dict["page"],
+            l=metadata_dict["l"],
+            t=metadata_dict["t"],
+        )
+
+    def to_metadata_dict(self) -> Dict[str, _OutlineMetadataKeys]:
+        """Convert object to a dict for storing as metadata on Document
+
+        Returns:
+            Dict[str, _OutlineMetadataKeys]: dict representation of object
+        """
+        return asdict(self)
+
```

Comment:
add a `class Outline` that basicaly just wraps a `List[OutlineItem]` so it's clear that hte List aspect is imprortant